llvm/include/llvm/SandboxIR/SandboxIR.h

```diff
@@ -743,6 +743,9 @@ class LoadInst final : public Instruction {
   static LoadInst *create(Type *Ty, Value *Ptr, MaybeAlign Align,
                           Instruction *InsertBefore, Context &Ctx,
                           const Twine &Name = "");
+  static LoadInst *create(Type *Ty, Value *Ptr, MaybeAlign Align,
+                          Instruction *InsertBefore, Context &Ctx,
+                          const Twine &Name = "", bool isVolatile);
```

Comment:
I would move the `isVolatile` flag before the `Ctx` argument.
Also we would need both versions of the create() function with the additional isVolatile argument + their bodies + tests :)

Reply:
> I would move the isVolatile flag before the Ctx argument.

will do

> Also we would need both versions of the create() function with the additional isVolatile argument + their bodies + tests :)

so for every function `create` function there needs to have a version where isVolatile is an argument?

Reply:
yes, 4 in total: 2 with isVolatile and 2 without.

Reply:
I think we can save some lines of code, do you think it'd be possible to just have 2 functions instead of 4. Where the isVolatile arg, has default value of `false` if there's no input? or should we stick to the original plan of having 4 functions?

Reply:
Makes sense. Something like `create(Type *Ty, Value *Ptr, MaybeAlign Align, Instruction *InsertBefore, Context &Ctx, bool IsVolatile = false, const Twine &Name = "")` ? I was suggesting using 4 functions because llvm::LoadInst is using separate constructors but I don't feel too strongly about it.

Reply:
correct me if im of course

Reply:
> Makes sense. Something like create(Type *Ty, Value *Ptr, MaybeAlign Align, Instruction *InsertBefore, Context &Ctx, bool IsVolatile = false, const Twine &Name = "") ?

exactly, so then we can just replace the `false` input with `IsVolatile` in the following code below
https://github.com/llvm/llvm-project/blob/173514ff129620364ac0ed61027e81194d997d76/llvm/lib/SandboxIR/SandboxIR.cpp#L464-L466

> I was suggesting using 4 functions because llvm::LoadInst is using separate constructors but I don't feel too strongly about it.

Yeah I figured, the only reason that'd consider adding 4 functions is to follow the same convention as what we're replicating. But I think we should be fine and it seems like it saves us some code too.

Reply:
do you approve? if so, i'll have to change the title to something more suited for our situation

Reply:
Yes, go for it.

Reply:
Either way is fine with me.

Reply:
@vporpo I'll have to close this PR unfortunately and ship the code to another PR. My local branch got corrupted and will create a mess with this PR if I tried to do a commit.